### DISCUSSION TITLE
Set up the upstream icons server

### DISCUSSION THREAD

**Discussion by aimuz** [dani-garcia/vaultwarden#1844] [Ideas]
In some cases vaultwarden is deployed in a network environment, this network environment may be limited by firewall reasons, such as unable to access services such as google.com, so that Icon cannot be downloaded, I think vaultwarden if you can add an upstream icons service settings, This service is responsible for downloading Icon. And vaultwarden is just a reverse proxy, or download the icon based on this

For example, icons.bitwarden.net is an official address.

**Comment by BlackDex** [comment]
The same goes for the bitwarden self-hosted, it also doesn't use the online service. We actually removed that because we do not want an external service to be used.

What you can do if you really want this, is to configure the reverse proxy to redirect those calls to the /icons/ to icons.bitwarden.com.
